### PR TITLE
feat: v0.1.3 working

### PR DESCRIPTION
**Pull request by andy8052**
feat: v0.1.3 working

### CODE REVIEW

File: contracts/StrategySushiswapPair.sol [hunk 1]
```diff
@@ -84,7 +84,7 @@ contract StrategySushiswapPair is BaseStrategy {
      * Provide an accurate expected value for the return this strategy
      * would provide to the Vault if `report()` was called right now
      */
-    function expectedReturn() public override view returns (uint256 _liquidity) {
+    function expectedReturn() internal view returns (uint256 _liquidity) {
```
Comment (Macarse) [inline contracts/StrategySushiswapPair.sol:87]: looks like you are not calling this anymore. You can remove the method.

File: contracts/StrategySushiswapPair.sol [hunk 2]
```diff
@@ -131,15 +131,16 @@ contract StrategySushiswapPair is BaseStrategy {
...
      */
-    function prepareReturn() internal override {
-        reserve = want.balanceOf(address(this)).sub(outstanding);
+    function prepareReturn(uint256 _debtOutstanding) internal override returns (uint256 _profit){
+        setReserve(want.balanceOf(address(this)).sub(_debtOutstanding));
         SushiChef(chef).deposit(pid, 0);
         uint _amount = IERC20(reward).balanceOf(address(this));
-        if (_amount < 1 gwei) return;
+        if (_amount < 1 gwei) return 0;
         swap(reward, token0, _amount / 2);
```
Comment (Macarse) [inline contracts/StrategySushiswapPair.sol:138]: SafeMath `div()`?

File: contracts/StrategySushiswapPair.sol [hunk 3]
```diff
@@ -131,15 +131,16 @@ contract StrategySushiswapPair is BaseStrategy {
...
         _amount = IERC20(reward).balanceOf(address(this));
         swap(reward, token1, _amount);
         add_liquidity();
+        _profit = want.balanceOf(address(this)).sub(getReserve()).sub(_debtOutstanding);
```
Comment (Macarse) [inline contracts/StrategySushiswapPair.sol]: mmmmm interesting.
I am doing `_profit = balanceOfWant().sub(getReserve());` without considering `_debtOutstanding`.

File: contracts/StrategySushiswapPair.sol [hunk 4]
```diff
@@ -173,24 +174,10 @@ contract StrategySushiswapPair is BaseStrategy {
      * Liquidate as many assets as possible to `want`, irregardless of slippage,
      * up to `_amount`. Any excess should be re-invested here as well.
      */
-    function liquidatePosition(uint256 _amount) internal override {
-        // TODO: Do stuff here to free up `_amount` from all positions back into `want`
+    function liquidatePosition(uint256 _amount) internal override returns (uint256 _amountFreed) {
+        uint256 before = want.balanceOf(address(this));
         SushiChef(chef).withdraw(pid, _amount);
```
Comment (Macarse) [inline contracts/StrategySushiswapPair.sol:178]: I think `_amount` might be greater than what you have available.

File: contracts/Vault.vy [hunk 5]
```diff
@@ -1,6 +1,41 @@
 #@version 0.2.7
-
-API_VERSION: constant(String[28]) = "0.1.2"
+"""
```
Comment (Macarse) [inline contracts/Vault.vy:2]: you shouldn't need to have the vault code in your repo.

File: contracts/StrategySushiswapPair.sol (hunk 4 above)
Comment (Macarse) [reply contracts/StrategySushiswapPair.sol:178]: nope! Vault.py was recently changed:
https://github.com/iearn-finance/yearn-vaults/blob/master/contracts/Vault.vy#L751

File: contracts/StrategySushiswapPair.sol (hunk 2 above)
Comment (Grandthrax) [reply contracts/StrategySushiswapPair.sol:138]: safe math just checks denom is not zero

File: contracts/StrategySushiswapPair.sol (hunk 3 above)
Comment (Grandthrax) [reply contracts/StrategySushiswapPair.sol]: yep debt outstanding does not impact profit. just reserve

File: contracts/StrategySushiswapPair.sol (hunk 4 above)
Comment (Grandthrax) [reply contracts/StrategySushiswapPair.sol:178]: But you are withdrawing an amount above your current balance. Why? Surely you want to withdraw _amount - before.

**Review by Grandthrax — COMMENTED** [review]
a few comments. I need to do a full review later

File: contracts/StrategySushiswapPair.sol (hunk 3 above)
Comment (andy8052) [reply contracts/StrategySushiswapPair.sol]: Ah good to know. Thanks

File: contracts/StrategySushiswapPair.sol (hunk 1 above)
Comment (andy8052) [reply contracts/StrategySushiswapPair.sol:87]: I am actually still using it [here](https://github.com/andy8052/strategy-sushi/pull/1/files#diff-31bca2dbef069066357afbbb8484cf779d5d14fbe57a0a63ab98e0dd4aeefc9bL120)

File: contracts/StrategySushiswapPair.sol (hunk 4 above)
Comment (andy8052) [reply contracts/StrategySushiswapPair.sol:178]: Yeah I think you are right @Grandthrax

**Pull request merged**